### PR TITLE
Add stop method

### PR DESCRIPTION
**Pull request by tfussell**
Part of https://github.com/giantswarm/operatorkit/pull/516

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add method to stop collector after is has been booted.
```
Comment (kopiczko) [inline CHANGELOG.md:14]:
```suggestion
- Add `Set.Stop` method to stop collector after is has been booted.
```

File: collector/set.go [hunk 2]
```diff
@@ -69,6 +69,29 @@ func (s *Set) Boot(ctx context.Context) error {
...
+
+		registered := prometheus.Unregister(s)
+		if !registered {
+			s.logger.LogCtx(ctx, "level", "debug", "message", "collector was not registered")
+		} else {
+			s.logger.LogCtx(ctx, "level", "debug", "message", "unregistered collector")
+		}
+	}
+
+	s.logger.Log("level", "debug", "message", "stopped collector")
```
Comment (kopiczko) [inline collector/set.go:90]:
```suggestion
	s.logger.LogCtx(ctx, "level", "debug", "message", "stopped collector")
```

File: collector/set.go [hunk 3]
```diff
@@ -69,6 +69,29 @@ func (s *Set) Boot(ctx context.Context) error {
...
+
```
Comment (kopiczko) [inline collector/set.go:91]: I'm not certain about the value of those logs but let's keep them if you think they are legit.

**Review by kopiczko — APPROVED** [review]

**Review by yulianedyalkova — APPROVED** [review]

File: collector/set.go (hunk 3 above)
Comment (tfussell) [reply collector/set.go:91]: It's pretty verbose, but this should only be executed rarely, and I could see it helping with future debugging. No strong feelings though.

**Pull request merged**